include/seastar/core/circular_buffer.hh

```diff
@@ -55,10 +56,10 @@ namespace seastar {
 ///     * pop_back() will invalidate end().
 ///
 /// reserve() may also invalidate all iterators and references.
-template <typename T, typename Alloc = std::allocator<T>>
+template <typename T, typename Alloc = std::allocator<std::remove_const_t<T>>>
```

Comment:
Ordinary containers don't allow const T:

https://godbolt.org/z/KMbhEYKKW

So I don't think we should either. Nor do I see any gain from it. The code that pops an io_request could pop it into a const variable.